mozdownload/scraper.py

```diff
@@ -431,7 +431,9 @@ def get_latest_build_date(self):
                          auth=self.authentication, headers=headers)
         r.raise_for_status()
 
-        return datetime.strptime(r.text.split('\n')[0], '%Y%m%d%H%M%S')
+        retval = datetime.strptime(r.text.split('\n')[0], '%Y%m%d%H%M%S')
+        r.close()
+        return relval
```

Comment:
This will still not work in case of exceptions before the `close()` call. A better approach here would have been a try/finally block. Same for the other replacement.